harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -957,6 +1002,73 @@ def test_legacy_checkpoint_loading(self, ckpt: str):
         assert state.batches_trained == 1, "batches_trained does not match"
         assert state.epochs_trained == 0, "epochs_trained does not match"
 
+    @pytest.mark.gpu
+    @pytest.mark.parametrize("aggregation_frequency", [1, 4])
+    def test_pytorch_11_const(self, aggregation_frequency: int, tmp_path: pathlib.Path):
+        checkpoint_dir = str(tmp_path.joinpath("checkpoint"))
+
+        config = utils.load_config(utils.tutorials_path("mnist_pytorch/const.yaml"))
+        hparams = config["hyperparameters"]
+
+        exp_config = utils.make_default_exp_config(
+            hparams,
+            scheduling_unit=1,
+            searcher_metric="validation_loss",
+            checkpoint_dir=checkpoint_dir,
+        )
+        exp_config.update(config)
+
+        example_path = utils.tutorials_path("mnist_pytorch/model_def.py")
+        example_context = utils.tutorials_path("mnist_pytorch")
+        trial_module = utils.import_module("MNistTrial", example_path, example_context)
+        trial_class = getattr(trial_module, "MNistTrial")  # noqa: B009
+        trial_class._searcher_metric = "validation_loss"
+
+        self.checkpoint_and_restore_no_callbacks(
+            trial_class=trial_class,
+            hparams=hparams,
+            tmp_path=tmp_path,
+            exp_config=exp_config,
+            steps=(1, 1),
+        )
+
+    @pytest.mark.gpu
+    @pytest.mark.PyTorch
+    @pytest.mark.parametrize("api_style", ["apex", "auto", "manual"])  # TODO: test apex
+    def test_pytorch_const_with_amp(self, api_style: str, tmp_path: pathlib.Path):
```

Comment:
what does this test offer that our other gpu amp unit tests don't?

I suspect, but don't recall exactly, that we wrote the old unit tests with the intention of turning them on in the future, and the e2e tests were just to have some coverage of the feature in the mean time.

Reply:
We can remove this if the e2e_tests were only a temporary solution, since `test_scaler_checkpointing_and_restoring` does the same thing but with extra metrics checking. Just need to fix the tests.

Reply:
removed